ci/test/lint-protobuf.sh

```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env bash
+
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+#
+# lint-protobuf.sh - lint the protobuf compatibility.
+
+set -euo pipefail
+
+. misc/shlib/shlib.bash
+
+BUILDKITE_REPO_REF="origin"
+./ci/test/utils/fetch-merge-target.sh
+
+ci_collapsed_heading "Lint protobuf"
+ci_try buf breaking src --against ".git#branch=$BUILDKITE_PULL_REQUEST_BASE_BRANCH,subdir=src"
```

Comment:
So we are checking protobuf compatibility against main branch? So on main branch this is probably a no-op.

Reply:
Added a check with https://github.com/MaterializeInc/materialize/pull/19157/commits/399fceece08d57c4fa79c0151664c8e4b8b9e3dc